js/node/src/session_options_helper.cc

```diff
@@ -23,21 +26,32 @@ void ParseExecutionProviders(const Napi::Array epList, Ort::SessionOptions &sess
   for (uint32_t i = 0; i < epList.Length(); i++) {
     Napi::Value epValue = epList[i];
     std::string name;
+    int deviceId = 0;
     if (epValue.IsString()) {
       name = epValue.As<Napi::String>().Utf8Value();
     } else if (!epValue.IsObject() || epValue.IsNull() || !epValue.As<Napi::Object>().Has("name") ||
                !epValue.As<Napi::Object>().Get("name").IsString()) {
       ORT_NAPI_THROW_TYPEERROR(epList.Env(), "Invalid argument: sessionOptions.executionProviders[", i,
                                "] must be either a string or an object with property 'name'.");
     } else {
-      name = epValue.As<Napi::Object>().Get("name").As<Napi::String>().Utf8Value();
+      auto obj = epValue.As<Napi::Object>();
+      name = obj.Get("name").As<Napi::String>().Utf8Value();
+      if (obj.Has("deviceId")) {
+        deviceId = obj.Get("deviceId").As<Napi::Number>();
+      }
     }
 
     // CPU execution provider
     if (name == "cpu") {
       // TODO: handling CPU EP options
     } else if (name == "cuda") {
-      // TODO: handling Cuda EP options
+      Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CUDA(sessionOptions, deviceId));
+    } else if (name == "dml") {
+#ifdef _WIN32
+      Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_DML(sessionOptions, deviceId));
+#else
+      ORT_NAPI_THROW_ERROR(epList.Env(), "DirectML EP is supported only on Windows");
```

Comment:
Technically the [DirectML nuget](https://www.nuget.org/packages/Microsoft.AI.DirectML/) contains `libdirectml.so` for WSL, and ORT targets Linux too, but we've never tested this combination of ORT+DML in WSL. So as you have it makes sense for now. (deferring to Yulong/Guenther for review)